### PR TITLE
Refactor o-message spacing and icon position.

### PR DESCRIPTION
**Pull request by notlee**
My aim was to fix the [margin issue reported here](#114). However it was
hard to update in isolation which lead to a broader spacing update and 
refactor. I didn’t want to spend too much time on it because we’ll be 
working on messaging more substantially soon. This should be a nice little
update in the meantime which solves [Olga's issue](#114).

- Uses o-spacing values. This increases the height of some messages.
- Adjusts icon placement.
- Keeps additional content to the same width as main message
content (instead of running under the close icon)
- Corrects message padding when actions move to a new line.
Previously the right margin used between the content and the
action could leave a large amount of whitespace to the
right of the message: #114
- There's less Sass now.

Reviewed by design:
https://financialtimes.slack.com/archives/C01481FKWA2/p1605632723106300

Fixes: #114

o-message-before-after-large-viewport
o-message-before-after-small-viewport

### CODE REVIEW

**Comment by origamiserviceuser** [comment]
o-message bundle size difference from 4.2.1 to 4f41436decf8c72f49c882e16c0c79347b7eab30
css, master: 0.81kb decrease (0.01kb/gzip)
css, internal: 1.69kb decrease (-0.05kb/gzip)
css, whitelabel: 0.64kb decrease (0.00kb/gzip)
An insignificant difference was also found for: js

**Comment by chee** [comment]
> css, master: 0.81kb decrease (0.01kb/gzip)
> css, internal: 1.69kb decrease (-0.05kb/gzip)
> css, whitelabel: 0.64kb decrease (0.00kb/gzip)

:)

File: src/js/construct-element.js [hunk 1]
```diff
@@ -11,7 +11,15 @@ export default {
 		}
 
 		messageElement.classList.add('o-message', `o-message--${opts.type}`, 'o-message--closed');
-		if (opts.inner) { messageElement.classList.add('o-message--inner'); }
+		if (!opts.close) {
+			// when close is disabled add the declarative close attribute
+			// which is used to apply style
+			messageElement.setAttribute('data-close', false);
+		}
+
```
Comment (chee) [inline src/js/construct-element.js:19]: is this attribute new? should we be prefixing it with `o-message` or anything?

**Review by chee — APPROVED** [review]
this looks good. there's a lot here, but i read it twice.

**Comment by origamiserviceuser** [comment]
o-message bundle size difference from 4.2.1 to 4f41436decf8c72f49c882e16c0c79347b7eab30
css, master: 0.81kb decrease (0.01kb/gzip)
css, internal: 1.69kb decrease (-0.05kb/gzip)
css, whitelabel: 0.64kb decrease (0.00kb/gzip)
An insignificant difference was also found for: js

**Comment by origamiserviceuser** [comment]
o-message bundle size difference from 4.2.1 to 1a7b48c31f1ed0e27e3f3b9582c28ea1805e8d4a
css, master: 0.78kb decrease (0.01kb/gzip)
css, internal: 1.66kb decrease (-0.05kb/gzip)
css, whitelabel: 0.61kb decrease (0.01kb/gzip)
An insignificant difference was also found for: js

File: src/js/construct-element.js (hunk 1 above)
Comment (notlee) [reply src/js/construct-element.js]: it is not a new attribute but I think we should be prefixing! but don't see it in the spec? [related discussion](https://github.com/Financial-Times/origami-website/issues/284)

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v4.2.2 :tada:

**Pull request merged**